roxie/ccd/ccdfile.cpp

```diff
@@ -1035,11 +1036,18 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
         Owned<IFile> local = createIFile(localLocation);
         if (traceRemoteFiles)
             DBGLOG("openFile adding file %s (localLocation %s)", lfn, localLocation);
-        bool isCompressed = selfTestMode ? false : pdesc->queryOwner().isCompressed();
+        IFileDescriptor &fdesc = pdesc->queryOwner();
+        bool isCompressed = selfTestMode ? false : fdesc.isCompressed();
         unsigned crc = 0;
         if (!selfTestMode)
             pdesc->getCrc(crc);
-        Owned<CRoxieLazyFileIO> ret = new CRoxieLazyFileIO(local.getLink(), size, modified, isCompressed, crc);
+        bool isKey = false;
+        IPropertyTree const & props = fdesc.queryProperties();
+        const char *kind = props.queryProp("@kind");
```

Comment:
minor: kind = fdesc.queryKind() is slightly better encapsulation (but the same code)